main.go

```diff
@@ -361,12 +368,18 @@ func requestTiles(w http.ResponseWriter, r *http.Request) error {
 	vars := mux.Vars(r)
 
 	sources := strings.Split(vars["name"], ",")
+	var extant []string
 	for _, source := range sources {
-		layer, err := requestTile(r, source)
-		if err != nil {
-			return err
+		if !slices.Contains(extant, source) {
+			layer, err := requestTile(r, source)
+			if err != nil {
+				return err
+			}
+			layers = append(layers, layer...)
+			extant = append(extant, source)
+		} else {
+			log.Debugf("Skipping duplicate layer %s in request %s", source, sources)
```

Comment:
Just out of curiosity - what security bug does this fix? Is it a DDOS from someone requesting a whole bunch of duplicate layers?

Reply:
Yeah, this was the one Checkmarx warning that everyone on Friday agreed was a real potential problem: a request like `/layername,layername,layername,layername,layername,layername/{z}/{x}/{y}.pbf` would kick off simultaneous DB queries for the same data.  There's no explicit limit to the length of a request, and AFAIK they can be thousands of characters long, so this creates a vulnerability to a not-even-distributed DoS attack.

Reply:
oh boy. Well, that's a good one to fix!